cljfmt/src/cljfmt/core.cljc

```diff
@@ -192,10 +192,37 @@
 (defn pattern? [v]
   (instance? #?(:clj Pattern :cljs js/RegExp) v))
 
-(defn- indent-matches? [key sym]
-  (cond
-    (symbol? key) (= key sym)
-    (pattern? key) (re-find key (str sym))))
+(defn root? [zloc]
+  (nil? (zip/up zloc)))
+
+(defn- top-level-form [zloc]
+  (->> zloc
+       (iterate z/up)
+       (take-while (complement root?))
+       last))
+
+(defn- ns-form? [zloc]
+  (some-> zloc
+          child-sexprs
+          first
+          #{'ns `ns}))
+
+(defn- ns-name-from-zloc [zloc]
```

Comment:
Better name would be `find-namespace`. Function should return a symbol. I don't know why you have `edn` or have that `if` in there.